### PR TITLE
Add an optional column with claim/unclaim/reclaim action to the test result page.

### PR DESCRIPTION
**Pull request by ChristianMa97**
Hey I´m Chris,
I was missing a claim option on the test result page and wanted to contribute it to the plugin.
Let me know if I should change something.

### Changes:
Old Test Result page:
image

New Test Result page:
image

image

image

When the user is logged in, it shows the option to claim/reclaim and drop the claim in a new "Claim-Action" column on the Test Result page. When the user is not logged in, these options are not displayed.
### Testing done

Tested the Plugin as anonymous and logged in user from an hpi:run instance.

### Jira Issues:
https://issues.jenkins.io/browse/JENKINS-4169?jql=resolution%20is%20EMPTY%20and%20component%3D15531

Quick Claim Action could be added in the same column:
https://issues.jenkins.io/browse/JENKINS-69430?jql=resolution%20is%20EMPTY%20and%20component%3D15531

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by Waschndolos** [comment]
+1 for this. This would certainly be helpful for Devs who want to fix a complete Test Class. Any comments from the Codeowners?

**Comment by Greybird** [comment]
Hi, sorry for the delay, will try to take a look in the following days !

**Comment by Greybird** [comment]
Hi,

I did a first run to get a first impression.
I found that when going to the page relative to a specific failed test result page, the table has no header, and that raising actions will fail
image

I did not debug further, will try to take a look in the next few days, but wanted to let you know if you would have a chance to take a look before I did.

**Comment by ChristianMa97** [comment]
Thank you for testing and the feedback, i'll look into it.

**Comment by ChristianMa97** [comment]
Hi,

Sorry for the delay, i addressed the issues, could you have a look at it again?

**Comment by Greybird** [comment]
Hi, 

I just did a small round of changes. Can you please take a look and confirm everything is working as expected ?

File: src/main/resources/hudson/plugins/claim/AbstractClaimBuildAction/tablerow.jelly [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
+            <j:if test="${it.canRelease()}">
+                <j:if test="${linkWritten}"><st:nbsp/></j:if>
+                <f:link href="claim/unclaim" post="true">
+                    <j:out value="${it.messageProvider.getUnclaimActionText()}"/>
+                </f:link>
+                <j:set var="linkWritten" value="true"/>
+            </j:if>
+            <j:if test="${it.canReassign()}">
+                <j:if test="${linkWritten}"><st:nbsp/></j:if>
+                <a id="reassign" href="#" onClick="ShowPopup(this); return false;">
```
Comment (ChristianMa97) [inline src/main/resources/hudson/plugins/claim/AbstractClaimBuildAction/tablerow.jelly:43]: I would add an \<br/> tag here for better formatting.
image
image

**Comment by ChristianMa97** [comment]
It´s looking good and everything I tried is working.

I added the configuration to the readme, feel free to adjust it.

**Comment by Greybird** [comment]
Thanks let's merge this!

**Comment by ChristianMa97** [comment]
My test-scope was to small when testing it. I found a bug when I tried it in production.
The functions to open and close the claimHoverPopup are always opening the popup of the first row.
I created a following PR to fix this #288

**Pull request merged**